lib/cc/analyzer/engines_config_builder.rb

```diff
@@ -11,11 +11,11 @@ class EnginesConfigBuilder
         :container_label,
       )
 
-      def initialize(registry:, config:, container_label:, source_dir:, include_paths:)
+      def initialize(registry:, config:, container_label:, source_dir:, include_paths_builder:)
```

Comment:
Also, IncludePathsBuilder does cache its own results, which is why we pass down the instance to EnginesConfigBuilder due to having the global excludes and then engine specific excludes.